packages/shared/src/components/MainLayout.tsx

```diff
@@ -138,15 +139,30 @@ export default function MainLayout({
     );
   };
 
+  const router = useRouter();
+  const feeds = Object.values(MainFeedPage);
+  const page = router?.route?.substring(1).trim() as MainFeedPage;
+  const isPageReady = isFeaturesLoaded && router.isReady;
+  const existingAnonymousUser =
+    firstVisit && new Date(firstVisit) < firstVisitRequirement;
   const shouldShowOverlay =
-    !user && isOnboardingOpen && onboardingV2 !== OnboardingV2.Control;
+    !user &&
+    onboardingV2 !== OnboardingV2.Control &&
+    (!page || feeds.includes(page)) &&
+    isPageReady &&
+    !existingAnonymousUser;
+
+  if (shouldShowOverlay) {
+    router.push('/onboard');
+  }
+
+  if (!isPageReady || shouldShowOverlay) return null;
```

Comment:
Just wanted to note that on any page, we are not showing anything, anymore until the page is ready. cc @rebelchris @idoshamun @capJavert @TomDowling

Reply:
Has some implications of how our first paint looks currently..
Before - some top skeleton:
<img width="1501" alt="image" src="https://github.com/dailydotdev/apps/assets/9803078/4f54d677-aaaf-413b-9a84-c99a0dbd5774">


After - just a blank screen:
<img width="1499" alt="image" src="https://github.com/dailydotdev/apps/assets/9803078/ea6ec494-e86a-4882-9a8b-2e1118fba9f6">

Reply:
@capJavert had to. The placement of the logo from other pages can be not consistent, like in this onboarding page.